app/models/grda_warehouse/tasks/hmis_csv_splitter.rb

```diff
@@ -0,0 +1,147 @@
+        next if filename.in?(manually_processed)
+
+        Rails.logger.debug "Splitting #{filename}"
+        source_file_path = File.join(source_path, filename)
+        destination_file_path = File.join(destination_path, filename)
+        next unless File.exist?(source_file_path)
+
+        results[filename] = { added: 0, original: 0 }
+        ::CSV.open(destination_file_path, 'wb') do |output|
+          ::CSV.foreach(source_file_path, **csv_options).each.with_index do |row, i|
+            results[filename][:original] += 1
+            output << row.headers if i.zero? # Include the header
+            # Add project limited
+            if filename.in?(project_related)
+              if row['ProjectID'].in?(project_ids)
+                output << row
+                results[filename][:added] += 1
+              end
+            elsif filename == 'Organization.csv'
+              if row['OrganizationID'].in?(organization_ids)
+                output << row
+                results[filename][:added] += 1
+              end
+            elsif filename == 'Client.csv'
+              if row['PersonalID'].in?(personal_ids)
+                output << row
+                results[filename][:added] += 1
+              end
+            else
+              # Add enrollment limited
+              if row['EnrollmentID'].in?(enrollment_ids) # rubocop:disable Style/IfInsideElse
+                output << row
+                results[filename][:added] += 1
+              end
+            end
+          end
+        end
+        Rails.logger.debug "Added #{results[filename][:added]} of #{results[filename][:original]} rows to #{filename}"
+      end
+      results
+    end
+
+    # Find relevant OrganizationIDs in Project.csv and make note
+    private def capture_relevant_organization_ids
+      ::CSV.foreach(File.join(source_path, 'Project.csv'), **csv_options).each do |row|
+        next unless row['ProjectID'].in?(project_ids)
+
+        organization_ids << row['OrganizationID']
+      end
+    end
+
+    # Find relevant EnrollmentID and PersonalIDs in Enrollment.csv and make note
+    private def capture_relevant_enrollment_ids
+      ::CSV.foreach(File.join(source_path, 'Enrollment.csv'), **csv_options).each do |row|
+        next unless row['ProjectID'].in?(project_ids)
+
+        enrollment_ids << row['EnrollmentID']
+        personal_ids << row['PersonalID']
+      end
+    end
+
+    private def csv_options
+      {
+        headers: true,
+        # header_converters: downcase_converter,
+        liberal_parsing: true,
+        encoding: 'iso-8859-1:utf-8',
+      }
+    end
+
+    private def downcase_converter
+      ->(header) { header.downcase }
+    end
+
+    private def manually_processed
+      [
+        'Export.csv',
+        'User.csv',
+      ]
+    end
+
+    private def project_related
+      [
+        'Project.csv',
+        'Inventory.csv',
+        'ProjectCoC.csv',
+        'Affiliation.csv',
+        'Funder.csv',
+      ]
```

Comment:
Should the 2024 file names be added here? Ce participation and hmis participation

Reply:
ProjectCoC too

Reply:
At the moment, this is 2022 only, I figured we'd update that after 10/1 or when we need to split a 2024 file.  That may be the only change necessary, but until we receive a 2024 file it felt premature.

Reply:
Ok. Yep only needed for 2022 for now.